client/state/data-layer/wpcom-http/pipeline/retry-on-failure/index.jsx

```diff
@@ -0,0 +1,108 @@
+/**
+ * External dependencies
+ */
+import {
+	get,
+	head,
+	sortBy,
+	toPairs,
+} from 'lodash';
+
+const requestQueue = new Map();
```

Comment:
> how is adding it and then stripping it off better?

I'm realizing now that this would never actually make it to Calypso right? We would be decorating the `WPCOM_HTTP_REQUEST` action object which I don't think gets past the data-layer.  

And in terms of what the benefit would be, heres a comparison:

1) retryCount Map:
- you need to construct a `key`.
- you need to keep track of request state and make sure to clear it at the right times. 
- you will need to write a clearMap() function _just for testing purposes_.
- in order to simulate an action being on its Nth attempt, you will need to call the handler N times.  ]

2) decorated in the action
- more declarative. just by looking at the action you can tell what retry number it is
- super easy to test. want to make sure that an action doesn't trigger a network request if it is at its maxRetry count? Just declaratively create the action in question!

With that said, i'll defer to your judgement 👍